File: go/test/faker_test.go
package test

import (
	"fmt"
	"gitlab.yunzhen-data.com/SharedComputing/Pir-SDK/go/lib"
	"testing"
	"time"
)

func TestFaker(t *testing.T) {

	sizePerItem := 10000
	start := time.Now()
	faker := pir.NewFaker()
	Ids := faker.GenerateCNIDCardNumber(sizePerItem)
	IdsByte := pir.ArrayOfVectorUInt8ToBytes(Ids)
	end := time.Now()
	fmt.Println(len(IdsByte))
	for i := 0; i < len(IdsByte); i++ {
		fmt.Println(string(IdsByte[i]))
	}
	fmt.Println("测试成功！")
	fmt.Println("耗时：", end.Sub(start))

}

File: java/src/test/java/com/pir/FakerTest.java
package com.pir;

import org.junit.Test;


public class FakerTest {
    static {
        System.load("/Users/admin/git/Pir-SDK/java/src/main/resources/libpir-java-arm64.dylib");
    }
    @Test
    public void testFaker() {
        int sizePerItem = 10000;
        long start = System.currentTimeMillis();
        Faker faker = new Faker();
        ArrayofVectorUInt8 ids = faker.GenerateCNIDCardNumber(sizePerItem);
        byte[][] idsByte = Vector2Byte.ArrayOfVectorUInt8ToBytes(ids);
        long end = System.currentTimeMillis();
        System.out.println(idsByte.length);
        for (byte[] idByte : idsByte) {
            System.out.println(new String(idByte));
        }
        System.out.println("测试成功！");
        System.out.println("耗时：" + (end - start) + " 毫秒");
    }



}
